File: codegen/results/v001/qwen3.5_27b_nothink/20260408_102056/python/008_prime_numbers/solution.py
import sys
import math

def sieve_of_eratosthenes(limit: int) -> None:
    if limit < 2:
        return

    # Boolean array where index represents the number
    # Initially assume all numbers are prime
    is_prime = [True] * (limit + 1)
    is_prime[0] = is_prime[1] = False

    # Optimization: iterate only up to sqrt(limit)
    for i in range(2, int(math.isqrt(limit)) + 1):
        if is_prime[i]:
            # Mark multiples of i starting from i*i
            for j in range(i * i, limit + 1, i):
                is_prime[j] = False

    # Print all numbers marked as prime
    for i in range(2, limit + 1):
        if is_prime[i]:
            print(i)

def main():
    if len(sys.argv) != 2:
        # If no argument is provided, do nothing or handle error as per spec
        # Spec says "Accept a single integer argument", implying we expect one.
        # If missing, we simply do nothing or exit.
        return
    
    try:
        n = int(sys.argv[1])
    except ValueError:
        # If the argument is not a valid integer, do nothing or exit
        return

    sieve_of_eratosthenes(n)

if __name__ == "__main__":
    main()

File: codegen/results/v001/qwen3.5_27b_nothink/20260408_102056/typescript/008_prime_numbers/solution.ts
// Accept N from command line arguments
const input = process.argv[2];

if (!input) {
  process.exit(0);
}

const n = parseInt(input, 10);

if (isNaN(n) || n < 2) {
  // No primes exist below 2
  process.exit(0);
}

const isPrime = (num: number): boolean => {
  if (num < 2) return false;
  if (num === 2) return true;
  if (num % 2 === 0) return false;

  const limit = Math.sqrt(num);
  for (let i = 3; i <= limit; i += 2) {
    if (num % i === 0) return false;
  }
  return true;
};

for (let i = 2; i <= n; i++) {
  if (isPrime(i)) {
    console.log(i);
  }
}